lib/wallet/wallet.js

```diff
@@ -1781,6 +1786,9 @@ class Wallet extends EventEmitter {
       if (!coin)
         continue;
 
+      if (acct && !await this.txdb.acctOwnsTX(acct, hash))
```

Comment:
It feels like the `Account` object should have a method called `hasTX`, making the `txdb` have a method `acctOwnsTX` doesn't feel like the best abstraction. Is it possible to return the account index on the `bid`? This would allow for something like:

```js
if (!own)
  continue;

if (acct && acct !== coin.accountId)
  continue;
```

Reply:
Also potentially the `coin` could have the account id on it. We need to be careful to not conflate the account id and the bip 44 account index, re: watch only wallet bug.

Reply:
We could, but the bids come from txdb serialized like so: 

https://github.com/pinheadmz/hsd/blob/90837d8fe46eade6e45b66cb3acb09aafa21de7c/lib/wallet/txdb.js#L3729-L3773

So adding an acct id to the bid would mean changing that serialization. No problem if you think that's best, but it'd be a breaking change.

Reply:
Same thing with txdb `coin`:

`this.own = false;`

We could change that from a bool to a byte (or how ever many bytes) to store account number?

Reply:
Is there a way to get account id by txid using the `txdb`? I think thats another solution.

Reply:
wallet layout has:
` *   T[account][hash] -> dummy (tx by account)`
... but nothing for _account by tx_

If it did, would that be cleaner than just getting a `has` bool from txdb?